### DISCUSSION TITLE
GitHub Pages autolinks fail

### DISCUSSION THREAD

**Discussion by bzliu94** [community/community#22154] [Other Feature Feedback, Questions, & Ideas]
<p>Hi,</p>
<p>I’m attempting to use autolinks on my Pages website; it’s not working for me.</p>
<p>Viewed via my <a href="http://github.io" rel="nofollow noopener">github.io</a> address, bare URL’s fail to automatically turn into links. But, when my Pages Markdown documents are viewed via browsing an appropriate repository through <a href="http://github.com" rel="nofollow noopener">github.com</a>, then the links show up. Am I missing something or is there a bug here (or is autolinking not supported for Pages webpages)? To be clear, by “autolinking”, I mean I am not using brackets or parentheses or any explicit Markdown notation.</p>
<p>Thanks,<br>
Brian</p>

**Comment by otter-computer** [comment]
<p>Hi, Brian, welcome to the GitHub Support Community!</p>
<p>Could you give us some more information about exactly what you mean? A link to the repository you’re working with, or some example source code, would be very helpful!</p>

**Comment by bzliu94** [comment]
<p>Hi Thomas,</p>
<p>Two links that show this are:<br>
<a href="https://bzliu94.github.io/" rel="nofollow noopener">https://bzliu94.github.io/</a><br>
<a href="https://github.com/bzliu94/bzliu94.github.io" rel="nofollow noopener">https://github.com/bzliu94/bzliu94.github.io</a></p>
<p>A blog entry that I have from August 29th, 2020 shows clickable automatically-generated links via the second address, but not via the first address.</p>
<p>I’d like to add links without using <code>[text](address)</code> syntax and would like to have links automatically-generated via, for example, text of form <code>http://www.domain.com/foo/bar</code>.</p>
<p>Thanks,<br>
Brian</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Thanks for dropping those links!</p>
<p>By default GitHub Pages uses Kramdown as the markdown processor for sites, whereas <a href="http://GitHub.com">GitHub.com</a> uses a variant of CommonMark—GitHub Flavoured Markdown. This is what’s causing your autolinks to work inside your repository but not on your Pages site.</p>
<p>To change this you’ll need to create a <code>_config.yml</code> file inside your repository with the following content to set your markdown renderer to GFM:</p>
<pre><code class="lang-auto">markdown: GFM
</code></pre>
<p><a href="https://docs.github.com/en/github/working-with-github-pages/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll" class="onebox" target="_blank">https://docs.github.com/en/github/working-with-github-pages/setting-a-markdown-processor-for-your-github-pages-site-using-jekyll</a></p>
<p>Once you’ve done that, and your site has had time to rebuild, any links you’ve got in there should be clickable just like inside your repository!</p>

**Comment by bzliu94** [comment]
<p>Hi Thomas,</p>
<p>Thanks for your help.</p>
<p>Brian</p>

**Comment by dotspencer** [comment]
<p>One thing to note is that you must include the <code>_config.yml</code> file in the same directory that your github pages are being served from. My repository was configured to serve the files from the <code>/docs</code> directory but the autolinks did not work until I moved the config file from the root to <code>/docs/_config.yml</code></p>
